code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -397,7 +405,8 @@
 	cyborg_base_icon = "engineer"
 	model_select_icon = "engineer"
 	model_traits = list(TRAIT_NEGATES_GRAVITY)
-	hat_offset = -4
+	hat_offset_x = list("north" = 0, "south" = 0, "east" = 4, "west" = -4, "northeast" = 4, "northwest" = -4, "southeast" = 4, "southwest" = -4)
+	hat_offset_y = list("north" = -4, "south" = -4, "east" = -4, "west" = -4, "northeast" = -4, "northwest" = -4, "southeast" = -4, "southwest" = -4)
```

Comment:
Byond doesn't like those macros in datum initializer since they are apparently non-const. Also they don't really avoid typos since lists will happily accept any typo'd macro for a key.
![image](https://github.com/user-attachments/assets/02228e68-b802-4936-8cd5-df398f9f776d)


I do sorta want to combine the x,y since that's how the 'rideable' component does it.
![image](https://github.com/user-attachments/assets/8a200c4e-ca9c-45ca-81c9-652dbff0ec5d)

Reply:
The macros are just `"[NORTH]"` that doesn't make any sense that it's non-constant

Reply:
"[NORTH]" evaluates to "1" or something doesn't it? So it is non-const. DM would need some kind of equivalent to c++ 'constexpr' probably.